crates/types/src/config/metadata_server.rs

```diff
@@ -58,8 +58,9 @@ pub struct MetadataServerOptions {
     /// Type of metadata server to start
     ///
     /// The type of metadata server to start when running the metadata store role.
+    // defined as Option<_> for backward compatibility with version < v1.2
     #[serde(flatten)]
-    pub kind: MetadataServerKind,
+    kind: Option<MetadataServerKind>,
```

Comment:
The weird thing I found is that even with `[serde(default)]` the decoder still complains about missing key `type`.